tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ControlSyntaxSuite.scala

```diff
@@ -1370,6 +1370,229 @@ class ControlSyntaxSuite extends BaseDottySuite {
+          Term.ApplyInfix(
+            Term.Match(
+              Term.Name("xs"),
+              List(
+                Case(Term.Name("Nil"), None, Lit.String("empty")),
+                Case(
+                  Pat.ExtractInfix(
+                    Pat.Var(Term.Name("x")),
+                    Term.Name("::"),
+                    List(Pat.Var(Term.Name("xs1")))
+                  ),
+                  None,
+                  Lit.String("nonempty")
+                )
+              )
+            ),
+            Term.Name("startsWith"),
+            Nil,
+            List(Lit.String("empty"))
+          ),
+          List(
+            Case(Lit.Boolean(true), None, Lit.Int(0)),
+            Case(Lit.Boolean(false), None, Lit.Int(1))
+          )
+        )
+      )
+    )
+
+  }
+
+  test("match-chained-complex-operator") {
+    runTestAssert[Stat](
+      """|val hello = xs match
+         |  case Nil => 0
+         |  case x :: xs1 => 1
+         |+ 1 match
+         |  case 1 => true
+         |  case 2 => false
+         |
+         |""".stripMargin,
+      assertLayout = None
+    )(
+      Defn.Val(
+        Nil,
+        List(Pat.Var(Term.Name("hello"))),
+        None,
+        Term.Match(
+          Term.Name("xs"),
+          List(
+            Case(Term.Name("Nil"), None, Lit.Int(0)),
+            Case(
+              Pat.ExtractInfix(
+                Pat.Var(Term.Name("x")),
+                Term.Name("::"),
+                List(Pat.Var(Term.Name("xs1")))
+              ),
+              None,
+              Term.Match(
+                Term.ApplyInfix(Lit.Int(1), Term.Name("+"), Nil, List(Lit.Int(1))),
+                List(
+                  Case(Lit.Int(1), None, Lit.Boolean(true)),
+                  Case(Lit.Int(2), None, Lit.Boolean(false))
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+
+  }
+
+  test("match-dot") {
+    val expected = Term.If(
+      Term.Match(
+        Term.Name("xs"),
+        List(
+          Case(Term.Name("Nil"), None, Lit.Boolean(false)),
+          Case(Pat.Wildcard(), None, Lit.Boolean(true))
+        )
+      ),
+      Lit.String("nonempty"),
+      Lit.String("empty")
+    )
+    runTestAssert[Stat](
+      """|if xs.match {
+         |  case Nil => false
+         |  case _ => true
+         |}
+         |then "nonempty"
+         |else "empty"
+         |""".stripMargin,
+      assertLayout = None
+    )(expected)
+
+    runTestAssert[Stat](
+      """|if xs.match
+         |  case Nil => false
+         |  case _ => true
+         |then "nonempty"
+         |else "empty"
+         |""".stripMargin,
+      assertLayout = Some(
+        """|if (xs match {
+           |  case Nil => false
+           |  case _ => true
+           |}) "nonempty" else "empty"
+           |""".stripMargin
+      )
+    )(expected)
+  }
+
+  test("match-dot-def") {
+    runTestAssert[Stat](
+      """|def mtch(x: Int): String =
+         |   x.match {
```

Comment:
+test as Significant Indentation(just remove {}) `(error: illegal start of statement ^.^trim())`

Reply:
Fixed!